library/core/src/ptr/mod.rs

```diff
@@ -2286,8 +2286,11 @@ impl<F: FnPtr> fmt::Debug for F {
 /// use `&raw const` instead.
 ///
 /// It is still an open question whether writing through an `addr_of!`-created pointer is permitted
-/// or not. Until that is decided, the same rules as for shared references apply: it is UB to write
-/// through a pointer created with this operation, except for bytes located inside an `UnsafeCell`.
+/// or not. Specifically, if the place `expr` evaluates to is based on a raw pointer, then the
+/// result of `addr_of!` inherits all permissions from that raw pointer. However, if the place is
+/// based on a reference, local variable, or `static`, then until all details are decided, the same
+/// rules as for shared references apply: it is UB to write through a pointer created with this
+/// operation, except for bytes located inside an `UnsafeCell`.
```

Comment:
Maybe it would make more sense to just state "the rules today are so and so, the pointer is always read-only unless the input is a raw pointer in which case permissions are preserved". And then say that these rules could be relaxed in the future.

Reply:
Not sure what that would improve, sounds equivalent.